workspaces/orchestrator/docs/orchestratorFormWidgets.md

```diff
@@ -705,34 +707,238 @@ This is different from `"ui:widget": "hidden"` which changes the widget type its
+
+```json
+{
+  "advancedPort": {
+    "type": "integer",
+    "ui:hidden": {
+      "when": "config.server.useDefaultPort",
+      "is": true
+    }
+  }
+}
+```
+
+### Complete Example
+
+```json
+{
+  "title": "Deployment Configuration",
+  "type": "object",
+  "properties": {
+    "deploymentType": {
+      "type": "string",
+      "title": "Deployment Type",
+      "enum": ["simple", "advanced", "custom", "managed"],
+      "default": "simple"
+    },
+    "environment": {
+      "type": "string",
+      "title": "Environment",
+      "enum": ["development", "staging", "production"],
+      "default": "development"
+    },
+    "replicas": {
+      "type": "integer",
+      "title": "Number of Replicas",
+      "default": 1,
+      "ui:hidden": {
+        "when": "deploymentType",
+        "is": "simple"
+      }
+    },
+    "customScript": {
+      "type": "string",
+      "title": "Custom Deployment Script",
+      "ui:widget": "textarea",
+      "ui:hidden": {
+        "when": "deploymentType",
+        "isNot": "custom"
+      }
+    },
+    "productionApprover": {
+      "type": "string",
+      "title": "Production Approver Email",
+      "ui:hidden": {
+        "anyOf": [
+          { "when": "environment", "isNot": "production" },
+          { "when": "deploymentType", "is": "simple" }
+        ]
+      }
+    },
+    "advancedSettings": {
+      "type": "object",
+      "title": "Advanced Settings",
+      "ui:hidden": {
+        "when": "deploymentType",
+        "is": ["simple", "managed"]
+      },
+      "properties": {
+        "cpu": {
+          "type": "string",
+          "title": "CPU Limit"
+        },
+        "memory": {
+          "type": "string",
+          "title": "Memory Limit"
+        }
+      }
+    }
+  }
+}
+```
+
+### Key Differences
 
 | Property                | Behavior                                    | Use Case                                                                               |
 | ----------------------- | ------------------------------------------- | -------------------------------------------------------------------------------------- |
 | `"ui:widget": "hidden"` | Changes widget type to hidden input         | Simple hidden form values                                                              |
 | `"ui:hidden": true`     | Keeps original widget but hides it visually | Hide widgets while preserving their functionality (e.g., ActiveText that fetches data) |
+| `"ui:hidden": {...}`    | Conditionally hides based on form data      | Dynamic forms that adapt to user input                                                 |
+
+### Behavior of Hidden Fields
+
+Hidden fields (regardless of hiding method):
 
-Hidden fields:
+- **Are not displayed** in the form
+- **Are not shown** in the wizard stepper navigation (multi-step forms)
+- **Still participate** in form validation
+- **Are included** in form submission
+- **Are excluded** from the review page
```

Comment:
Thanks for stating that explicitly in the documentation, this might be source of future confusion.

Regarding the Review step, what about mentioning that there are hidden fields not listed here (on the Review page) but will be part of the request?

Eventually, in a future PR, we can make their list available for the user (once we get such a requirement).

Reply:
<img width="1507" height="687" alt="Screenshot 2025-12-16 at 12 46 08 PM" src="https://github.com/user-attachments/assets/80cd783d-f7de-4fdc-bf23-83b1f8b778c3" />
<img width="1476" height="646" alt="Screenshot 2025-12-16 at 12 45 50 PM" src="https://github.com/user-attachments/assets/bbac2dc4-2a25-48fa-ac5d-7dbb08a2878c" />

if there are hidden field, an info alert will be displayed in review step.